src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Core/SymbolIsBannedAnalyzerBase.cs

```diff
@@ -156,19 +150,62 @@ private void OnCompilationStart(CompilationStartAnalysisContext compilationConte
 
             return;
 
+            bool IsBannedSymbol([NotNullWhen(true)] ISymbol? symbol, [NotNullWhen(true)] out BanFileEntry? entry)
+            {
+                if (symbol is { ContainingSymbol.Name: string parentName } &&
+                    bannedApis.TryGetValue((parentName, symbol.Name), out var entries))
+                {
+                    foreach (var bannedFileEntry in entries)
+                    {
+                        foreach (var bannedSymbol in bannedFileEntry.Symbols)
```

Comment:
we defer creating the 'Symbols' for an entry until the point it is actually needed.  this is hte majority of hte perf in here.